// pagination-js/dist/pagination.js
"use strict";

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.getPagination = getPagination;

var _constant = require("./constant");

/**
 * Return the pagination Object
 * @param query
 * @param totalItem
 * @returns {{totalItem: *, page: *, itemPerPage: *, minIndex: number, maxIndex: number, hasNextPage: boolean, hasPrePage: boolean}}
 */
function getPagination(query, totalItem) {
    // Default page = 1
    var currentPage = query.page === undefined ? _constant.DEFAULT_CURRENT_PAGE : parseInt(query.page);
    // Default item_per_page = 10
    var itemPerPage = query.itemPerPage === undefined ? _constant.DEFAULT_ITEM_PER_PAGE : parseInt(query.itemPerPage);

    var minIndex = (currentPage - 1) * itemPerPage;
    var maxIndex = currentPage * itemPerPage;
    var totalPage = totalItem / itemPerPage;
    var hasNextPage = totalPage > currentPage;
    var hasPrePage = currentPage > 1;
    return {
        totalItem: totalItem,
        page: currentPage,
        itemPerPage: itemPerPage,
        minIndex: minIndex,
        maxIndex: maxIndex,
        hasNextPage: hasNextPage,
        hasPrePage: hasPrePage
    };
} /**
   * Created by Tien Nguyen on 12/21/16.
   */
